### PR TITLE
Move download script to root directory

### LINKED ISSUE #2: Support tests for multimodel statistics preprocessor function
To support testing the multimodel statistics preprocessor function with the data from this repository, the following still needs to be done

**Importable data loader #3:**
- [x] A function for loading the data conveniently, e.g. `load_timeseries()` which returns a cubelist (use [`cube_helper.load`](https://cube-helper.readthedocs.io/en/latest/Cube%20Helper%20Module.html#cube_helper.load) for loading the data?)
- [x] Make installable module `esmvaltool_sample_data`

**Download script in #4:**
- [x] Replace the `.` in the paths with a `/` for all subdirectories of esmvaltool_sample_data/data/timeseries
- [x] Download all timeseries datasets, but only save small ones (e.g. skip AWI data for now, because they save one file per year and this becomes big even with subsetting)
- [x] Improve download script so the subsetting is done for equal geographical regions/vertical levels instead of just by slicing the first 2 steps
- ~Select also an ocean variable for subsetting (item 2 in #1), e.g. with a daily frequency (if multimodel statistics supports it)~ -> this needs proper support from iris first
- [x] Clean up the script for downloading data (now in `esmvaltool_sample_data/sample_data.py`)

**Add metadata #11 :**
- [x] Add license information
- [x] Improve README
- [x] Improve CONTRIBUTING
- [x] Update `setup.py`

**Other:**
- [x] #9 Rename download script to `download_sample_data.py` and move it to root

### PR DESCRIPTION
**Pull request by stefsmeets**
This PR moves the download script to the root directory and gives it a more logical name.

### CODE REVIEW

File: download_sample_data.py [hunk 1]
```diff
@@ -283,7 +278,6 @@ def main():
 
     The resulting list of files is then sampled and stored locally in the
     directory 'data' using a commonly used directory structure.
-
     """
     cfg_file = Path(__file__).parent.parent / "config.yml"
```
Comment (nielsdrost) [inline download_sample_data.py]:
```suggestion
    cfg_file = Path(__file__).parent / "config.yml"
```

Comment (nielsdrost) [reply download_sample_data.py]: And: facets_file = Path(__file__).parent / "esmvaltool_sample_data" / "datasets.yml"

**Comment by nielsdrost** [comment]
I tried to run this script but ran into some trouble.

- This error https://github.com/ESGF/esgf-pyclient/issues/57, solved by adding a few lines at the top to set the SSL version (see issue)
- The config file and dataset.yml locations do not seem to be correctly specified
- I get a "pyesgf.search.exceptions.EsgfSearchException: Error loading available shards" error.

Then I gave up for today ;-)

**Comment by stefsmeets** [comment]
> 
> 
> I tried to run this script but ran into some trouble.
> 
>     * This error [ESGF/esgf-pyclient#57](https://github.com/ESGF/esgf-pyclient/issues/57), solved by adding a few lines at the top to set the SSL version (see issue)
> 
>     * The config file and dataset.yml locations do not seem to be correctly specified
> 
>     * I get a "pyesgf.search.exceptions.EsgfSearchException: Error loading available shards" error.
> 
> 
> Then I gave up for today ;-)

Thanks @nielsdrost , good points and it seems like I should have tested this myself first 😅

**Comment by stefsmeets** [comment]
The `esgf` issues I cannot help with, but I can make sure that `datasets.yml` is correctly referenced. Should work as intended now (or at least it does so on my pc).

**Review by bouweandela — CHANGES_REQUESTED** [review]
The download script and its configuration file (datsets.yml) belong together, so if you're moving one to the top level, I would also move the other to the top level.

**Comment by stefsmeets** [comment]
Hi @bouweandela , can you have another look? I think this is ready to merge!

File: download_sample_data.py [hunk 2]
```diff
@@ -283,13 +278,13 @@ def main():
...
-
     """
-    cfg_file = Path(__file__).parent.parent / "config.yml"
+    project_dir = Path(__file__).parent
+    cfg_file = project_dir / "config.yml"
     with cfg_file.open() as file:
         cfg = yaml.safe_load(file)
 
-    facets_file = Path(__file__).parent / "datasets.yml"
+    facets_file = project_dir / 'esmvaltool_sample_data' / "datasets.yml"
```
Comment (bouweandela) [inline download_sample_data.py]: when I try to run the script, I get
```
$ python download_sample_data.py 
Traceback (most recent call last):
  File "download_sample_data.py", line 326, in <module>
    main()
  File "download_sample_data.py", line 288, in main
    with facets_file.open() as file:
  File "/home/bandela/conda/envs/esmvaltool_test_data/lib/python3.8/pathlib.py", line 1221, in open
    return io.open(self, mode, buffering, encoding, errors, newline,
  File "/home/bandela/conda/envs/esmvaltool_test_data/lib/python3.8/pathlib.py", line 1077, in _opener
    return self._accessor.open(self, flags, mode)
FileNotFoundError: [Errno 2] No such file or directory: 'esmvaltool_sample_data/datasets.yml'
```
I think you may have to undo this change

**Review by bouweandela — COMMENTED** [review]
Thanks for making the changes, I tried to run the script but it failed on loading the configuration file. I also noticed that the documentation on how run the script in CONTRIBUTING.md has not been updated yet.

**Comment by stefsmeets** [comment]
> 
> 
> Thanks for making the changes, I tried to run the script but it failed on loading the configuration file. I also noticed that the documentation on how run the script in CONTRIBUTING.md has not been updated yet.

Yeah, my bad for not testing if it worked. Should be good now 👍

**Review by bouweandela — APPROVED** [review]

**Pull request merged**